### PR TITLE
Fixes #25340 - improve SSL error exception

### PR DESCRIPTION
**Pull request by lzap**
SSIA

### CODE REVIEW

**Comment by lzap** [comment]
Random failure, RTM.

File: lib/proxy_api/resource.rb [hunk 1]
```diff
@@ -122,6 +122,9 @@ def ssl_auth_params
         :ssl_ca_file      =>  ca_cert,
         :verify_ssl       =>  OpenSSL::SSL::VERIFY_PEER
       }
+    rescue StandardError => exception
+      Foreman::Logging.exception("Unable to read SSL certification or key for proxy communication", exception)
+      raise Foreman::Exception.new(N_("Unable to read SSL certification or key for proxy communication: %s"), exception.message)
```
Comment (tbrisker) [inline lib/proxy_api/resource.rb]: Perhaps add info about which settings should be checked? something like `Please check settings for ssl_certificate, ssl_ca_file and ssl_priv_key and ensure they are readable by the foreman user.`?

**Review by tbrisker — CHANGES_REQUESTED** [review]

File: lib/proxy_api/resource.rb (hunk 1 above)
Comment (lzap) [reply lib/proxy_api/resource.rb]: Done.

**Review by tbrisker — APPROVED** [review]
Thanks @lzap !

**Comment by tbrisker** [comment]
4b1e14290 - 1.20-stable

**Pull request merged**